app/controllers/candidate_interface/adviser_sign_ups_controller.rb

```diff
@@ -1,33 +1,41 @@
 module CandidateInterface
   class AdviserSignUpsController < CandidateInterfaceController
-    before_action :set_adviser_sign_up
     before_action :render_404_unless_available
 
-    def new; end
+    def show
+      @adviser_sign_up = Adviser::SignUp.build_from_hash(application_form, params[:preferred_teaching_subject_id])
+    end
 
-    def create
-      if @adviser_sign_up.save
-        flash[:success] = t('application_form.adviser_sign_up.flash.success')
+    def new
+      @adviser_sign_up = Adviser::SignUp.build_from_hash(application_form, params[:preferred_teaching_subject_id])
+    end
 
-        track_adviser_sign_up
+    def continue
+      @adviser_sign_up = Adviser::SignUp.new(adviser_sign_up_params.merge(application_form:))
 
-        redirect_to candidate_interface_details_path
+      if @adviser_sign_up.valid?
+        redirect_to candidate_interface_adviser_sign_up_path(application_form.id, preferred_teaching_subject_id: @adviser_sign_up.preferred_teaching_subject_id)
```

Comment:
```suggestion
        redirect_to candidate_interface_adviser_sign_up_path(preferred_teaching_subject_id: @adviser_sign_up.preferred_teaching_subject_id)
```

I'm pretty sure you don't need the `application_form_id` here. I believe (and I could be wrong) that everything in the `candidate_interface` is scoped by default to the `current_application_form`. But if this breaks things, ignore me.

Reply:
Somehow I get the param missing error if I remove it 🤔